### PR TITLE
Fixed four bugs in webservice class

### PR DESCRIPTION
**Pull request by dasu2013**
Hi Igor,

I fixed following four issues in webservice class.

Recurrence meeting didn't work. ZOOM_RECURRING_MEETING should be directly assign into request object as a type property ( $data['type'] = ZOOM_RECURRING_MEETING;).
Updated the http method in meeting update to 'PATCH'. It is not 'POST' in API version 2 (https://zoom.github.io/api/#update-a-meeting)
In Zoom API version 2 'alternative host' has moved inside settings. Therefore we need that in '_make_meeting_data_v2' method.
There is an error in 'user_getbyemail' method. It is used get all the users as a list using 'users/' API call. However this returns only 30 records bydefault and 300 maximum. So this is not correct way to get user by email address. We can use this 'https://api.zoom.us/v2/users/{userId}' instead of that.
Dasu

### CODE REVIEW

**Comment by tigusigalpa** [comment]
Hi!
I dont remember is my master branch up to date with branch zoomapiv2 branch? The changes was pulled to master, is it possible to pull it to zoomapiv2?

**Review by tigusigalpa — APPROVED** [review]

**Review by tigusigalpa — APPROVED** [review]

**Comment by tigusigalpa** [comment]
Your commits are good, but you forked files from old master not zoomapiv2 so for now is a lot of conflicts..

File: classes/webservice.php [hunk 1]
```diff
@@ -380,21 +380,18 @@ public function user_getbyemail($email) {
...
-                                $this->current_user = $apiuser;
-                            }
-                            $this->lastresponse = $apiuser;
-                            return $apiuser;
-                        }
+                $url = 'users/'.$email;
+                try {
+                    $user = $this -> make_call($url);
+                }catch (moodle_exception $e) {
+                    require_once($CFG->dirroot.'/mod/ncmzoom/lib.php');
```
Comment (tigusigalpa) [inline classes/webservice.php:387]: mod/ncmzoom ???

Comment (Dasuntha) [reply classes/webservice.php:387]: Sorry This is my mistake. It should be zoom not ncmzoom.

**Comment by Dasuntha** [comment]
Hi Igor,

Is it possible to update your master branch into latest version? Then I can fork it and make another version. 

Dasu

**Comment by tigusigalpa** [comment]
hi! maybe you will fork **zoomapiv2** and push it there? or its unpossible?

**Pull request closed without merging**